src/metabase/task/QUARTZ.md

```diff
@@ -0,0 +1,153 @@
+# Quartz Scheduler in Metabase
+
+This document provides guidance for developers on how to use the Quartz scheduler in Metabase for background task scheduling.
+
+## Overview
+
+Metabase uses [Quartz](http://www.quartz-scheduler.org/) for scheduling and executing background tasks. The Quartz scheduler is initialized during Metabase startup and manages various tasks like syncing databases, sending pulses, cleaning up sessions, and more.
+
+In Quartz, there are two fundamental concepts:
+
+- **Job**: Represents a unit of work to be executed. Jobs are defined as classes that implement specific interfaces and contain the actual code to be run.
+- **Trigger**: Defines when a job should be executed - it can be based on a schedule (like a cron expression), a specific time, or in response to events.
+
+Jobs and triggers are created separately and then associated with each other when scheduled in the Quartz system.
+
+## Use cases at Metabase
+
+This list is not complete and is open to additions. If you have other use cases or patterns for Quartz tasks in Metabase, please feel free to add them here.
+
+### 1. One-Time Startup Tasks
+
+These tasks execute once when Metabase starts up. They're useful for initialization operations.
```

Comment:
I think it's worth emphasizing that it runs on every instance when starting up a multi-instance deployment (e.g. every cloud "instance"), and if the process gets restarted (e.g. OOM).

Going forward I don't think we want to keep using Quartz for this kind of trivial stuff that doesn't need a calendar or across-instance coordination. See: https://linear.app/metabase/issue/ADM-465/stop-using-quartz-to-run-index-update-worker

Reply:
good point, added the note and also alternative